toughday/src/main/java/com/adobe/qe/toughday/internal/core/config/parsers/yaml/YamlBuilder.java

```
@@ -0,0 +1,95 @@
+        representer.addClassTag(YamlDumpConfiguration.class, Tag.MAP);
+        for (Class type : ReflectionsContainer.getSubTypesOf(YamlDumpAction.class)) {
+            representer.addClassTag(type, Tag.MAP);
+        }
+
+    }
+
+    private static void buildInstance() {
+        DumperOptions dumperOptions = getDumperOptions();
+
+        // Configure the representer to ignore empty fields when dumping the object. By default, each empty filed is represented as {}.
+        Representer representer = getRepresenter();
+        configureYamlRepresenterToExcludeClassTags(representer);
+
+        Constructor constructor = new Constructor(GenerateYamlConfiguration.class);
+        instance = new Yaml(constructor, representer, dumperOptions);
+    }
+
+    /**
+     * Getter for the Yaml instance.
+     */
+    public static Yaml getYamlInstance() {
+        if (instance == null) {
+            buildInstance();
+        }
+
+        return instance;
+    }
+
+
+    private static Representer getRepresenter() {
+        return new Representer() {
+            @Override
+            protected NodeTuple representJavaBeanProperty(Object javaBean, Property property, Object propertyValue, Tag customTag) {
+
+                Method method = null;
+                try {
+                    if (propertyValue == null) {
+                        propertyValue = "";
+                    }
+                    method = propertyValue.getClass().getMethod("isEmpty");
+                } catch (NoSuchMethodException ignored) { }
+
+                if (method == null) {
+                    return super.representJavaBeanProperty(javaBean, property, propertyValue, customTag);
+                } else {
+
+                    try {
+                        if (Boolean.valueOf(method.invoke(propertyValue).toString())) {
+                            return null;
+                        }
+                    } catch (Throwable t) {
+                        t.printStackTrace();
```

Should we do something else here beside printing the stack trace?

---

This error should not impact the overall functionality. It would only mean that the empty collections are also included in the Yaml representation on an object so I believe we can simply ignore this exception.